rllib/examples/noveld_exploration.py

```diff
@@ -0,0 +1,89 @@
+"""Example for using NovelD exploration for sparse reward settings.
+
+Note: NovelD works to this point only synchronously, i.e. the 
```

Comment:
How can we change this? What do we need to make this work in distributed case?

Reply:
@ArturNiederfahrenhorst I talked with @sven1977 about this and this should be the next step. In short, the updates of the NovelD network have to occur more often (namely in `postprocess_trajectory`. If now workers collect samples in parallel all these updates have to be synched. Theoretically, this will be something in in DDPPO. 
I am open for discussion (with a beer :))

Reply:
Great question @ArturNiederfahrenhorst , and yeah, this is currently a somewhat unsolved problem: For curiosity + PPO, you need to more frequently update the ICM model (currently done inside the workers' postprocess trajectory after every(!) rollout) than the actual (e.g. PPO) policy model (only done when entire train batch is completely collected from all remote workers).

Once we have Ray Train running inside RLlib and therefore RLlib will have a simple "learner API" available in its `training_iteration` methods (and even on the rollout workers), then we can call these "sub-updates" of the ICM model easily from within the RolloutWorkers.

Sort of like this:
<img width="1762" alt="image" src="https://user-images.githubusercontent.com/5839854/170244394-37a56498-ae57-4a59-98e2-49948477fdee.png">

Reply:
Nice sketch @sven1977 - that makes it very clear. 

Waiting for `trainer_train` then ...